File: app/src/main/java/dev/pitlor/smssync/db/daos/SyncDao.java
package dev.pitlor.smssync.db.daos;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.time.OffsetDateTime;

import dev.pitlor.smssync.db.entities.Sync;

@Dao
public interface SyncDao {
    @Query("SELECT date FROM sync ORDER BY date LIMIT 1")
    LiveData<OffsetDateTime> getLastSync();

    @Insert
    void addSync(Sync date);
}

File: app/src/main/java/dev/pitlor/smssync/datasources/daos/SyncDao.kt
package dev.pitlor.smssync.datasources.daos

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import dev.pitlor.smssync.datasources.Sync
import java.time.OffsetDateTime

@Dao
interface SyncDao: BaseDao<Sync> {
    @Query("SELECT * FROM sync ORDER BY startTime DESC LIMIT 1")
    fun getLastSync(): LiveData<Sync?>
}